olive/workflows/run/config.py

```diff
@@ -49,10 +52,40 @@ class RunConfig(ConfigBase):
     verbose: bool = False
     input_model: ModelConfig
     systems: Dict[str, SystemConfig] = None
+    data_config: Dict[str, DataConfig] = {
+        DefaultDataContainer.DATA_CONTAINER.value: DataConfig(),
+        DEFAULT_HF_DATA_CONTAINER_NAME: DataConfig(
+            name=DEFAULT_HF_DATA_CONTAINER_NAME,
+            type=HuggingfaceContainer.__name__,
+        ),
+    }
     evaluators: Dict[str, OliveEvaluatorConfig] = None
     engine: RunEngineConfig
     passes: Dict[str, RunPassConfig]
 
+    @validator("data_config", pre=True, each_item=True, always=True)
+    def validate_data_config(cls, v, values):
+        hf_config = values["input_model"].dict()["config"].get("hf_config", {})
```

Comment:
Should we make the `data_config` interact with the `input_model`'s `hf_config`? I think this makes the relationship complicated since the hf_config overrides the data_config but there is no clear reason why the input_model.hf_config should have a `dataset` member to begin with. 

Personally, I think keeping the data_config as an independent component is less complicated. And we only use the `input_model.hf_config` to load the model (and, in the future, get input_names, output_names, dynamic_axes automatically) and nothing else.

Reply:
In this case, we would have model name and task repeated for the `data_config` and `input_model.hf_config` which I think is okay since they are used for different purposes. In the validator, we can check that the values are same to make sure the user didn't make a mistake.

Reply:
Marking as resolved as discussed offline. We can revisit this topic later.